### PR TITLE
Early Exit Graphs/Nodes

### LINKED ISSUE #37: Early exit node/graph
**Is your feature request related to a problem? Please describe.**
Ability to cleanly exit a node or entire build.

**Describe the solution you'd like**
Lucas:
> Something like `return` in the code block to early exit the node's execution.
   Something like `exit()` or `halt()` to gracefully stop the graph from executing.
   Currently these can be handled with crazy indentation and raising an exception (respectively).

Michael:
> I like something explicitly nxt oriented for stopping the full graph. like `graph_stop()`. What are the ramifications for sub graphs? Does this only halt local subgraph or full super graph? Additionally can there(should there?) be ways to skip hierarchy `stop_siblings()` or `skip_descendants()`?

### PR DESCRIPTION
**Pull request by ImLucasBrown**
`+` Added exceptions for gracefully exiting graphs/nodes. `ExitNode` and `ExitGraph.`
`*` New exceptions are directly available in the globals, just call `raise ExitGraph`, a message can also be included `raise ExitNode("Goodbye")`.
`...` Added unittest for new exceptions.

---

Fixes #37

### CODE REVIEW

File: nxt/stage.py [hunk 1]
```diff
@@ -4046,13 +4060,16 @@ def run(runtime_layer, stage=None, rt_node=None, custom_code=None):
...
+    exit_exception = None
     try:
         console.runcode(_code)
     except GraphError:
         if not console.run_as_global:
             clean_globals(lines, good_keys, console.globals)
         runtime_layer.running = False
         raise
+    except (ExitNode, ExitGraph) as exit_exception:
+        pass
```
Comment (MichaelAldrich) [inline nxt/stage.py]: Did you experiment with reworking this whole try?
Something like this I think would clean up some of the existing mess as well. What do you think?
```
try:
    console.runcode(_code)
except (GraphError, ExitNode, ExitGraph):
    raise
finally:
    runtime_layer.running = False
    if not console.run_as_global:
    ...
```

File: nxt/builtin/sub_graphs.nxt [hunk 2]
```diff
@@ -34,8 +34,10 @@
                 "            continue",
                 "        attr_path = nxt_path.make_attr_path(nxt_path.WORLD, attr_name)",
                 "        params[attr_path] = attr_val",
-                "    ",
-                "    result_rt_layer = sub_stage.execute(parameters=params, start=self._start)",
+                "    try:",
+                "        result_rt_layer = sub_stage.execute(parameters=params, start=self._start)",
+                "    except ExitGraph as exit_graph:",
+                "        result_rt_layer = exit_graph.runtime_layer",
```
Comment (MichaelAldrich) [inline nxt/builtin/sub_graphs.nxt:40]: A bit picky, but can we get a simple test for this too?

File: nxt/stage.py (hunk 1 above)
Comment (ImLucasBrown) [reply nxt/stage.py]: Yeah I thought of this, I'm not convinced we want to do the cache value comparison (LN 4077) in the event of a `GraphError`. I can't remember if we had a good reason for _not_ doing that.

File: nxt/builtin/sub_graphs.nxt (hunk 2 above)
Comment (ImLucasBrown) [reply nxt/builtin/sub_graphs.nxt:40]: Shoot, I said I'd do that and never did. I'll whip up a test.

**Pull request merged**